File: backend/src/stock.cpp
#include <iostream>
#include "stock.h"

// Parameterised constructor
Stock::Stock(std::string name, std::string ticker, float price) {
    this->name = name;
    this->ticker = ticker;
    this->price = price;
    std::cout << "Stock created with values" << std::endl;
}

void Stock::printStock() {
    std::cout << "Name: " << name << std::endl;
    std::cout << "Ticker: " << ticker << std::endl;
    std::cout << "Price: " << price << std::endl;
}

// Update price
float Stock::getPrice() { return price; }
void Stock::setPrice(float price){ this->price = price; }

File: backend/src/main/java/dev/elijahao/stock_trading_exchange/Stock.java
package dev.elijahao.stock_trading_exchange;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "stocks")
@Data
@NoArgsConstructor
@AllArgsConstructor

public class Stock {
    @Id
    private String ticker;
    private String name;
    private float price;

    public void printStock() {
        System.out.println("Ticker: " + this.ticker);
        System.out.println("Name: " + this.name);
        System.out.println("Price: " + this.price);
    }
    
}
